mirrord/progress/src/lib.rs

```diff
@@ -106,6 +106,16 @@ pub trait Progress: Sized {
     fn set_done(&mut self, msg: Option<&str>, failure: bool);
 }
 
+pub struct NoProgress;
```

Comment:
Technically no mostly using it in operator, I wanted to implement progress on `()` but then decided to make it a custom type to prevent any future misunderstandings